lib/net.js

```diff
@@ -669,14 +669,12 @@ Socket.prototype._destroy = function(exception, cb) {
 };
 
 Socket.prototype._getpeername = function() {
-  if (!this._peername) {
-    if (!this._handle || !this._handle.getpeername) {
-      return {};
-    }
-    const out = {};
-    const err = this._handle.getpeername(out);
-    if (err) return {};  // FIXME(bnoordhuis) Throw?
-    this._peername = out;
+  if (!this._handle || !this._handle.getpeername) {
+    return this._peername || {};
+  } else if (!this._peername) {
+    this._peername = {};
+    // FIXME(bnoordhuis) Throws when returns not 0?
```

Comment:
```suggestion
    // FIXME(bnoordhuis) Throw when the return value is not 0?
```
:-) (and ditto on line 712)